src/hotkey/hotkey_item.cpp

```diff
@@ -293,9 +293,9 @@ bool hotkey_keyboard::matches_helper(const SDL_Event& event) const
 	const hotkey_command& command = get_hotkey_command(get_command());
 
 	if((event.type == SDL_KEYDOWN || event.type == SDL_KEYUP)
-		&& (mods & KMOD_CTRL || mods & KMOD_ALT || mods & KMOD_GUI || command.toggle || CKey::is_uncomposable(event.key))
+		&& (mods == KMOD_NONE || mods & KMOD_CTRL || mods & KMOD_ALT || mods & KMOD_GUI || command.toggle || CKey::is_uncomposable(event.key))
```

Comment:
This line seems strange, it now seems to accept any combination of modifies, including no modifiers, except when the modifiers are exactly Shift.

Reply:
As far as I can understand, SDL_TEXTINPUT is handled [here](https://github.com/wesnoth/wesnoth/blob/master/src/hotkey/hotkey_item.cpp#L301) to make shifted hotkeys work?